checks/trademarks.php

```diff
@@ -0,0 +1,287 @@
+		'disqus-',
+		'divi-',
+		'dropbox-',
+		'easy-digital-downloads-',
+		'elementor-',
+		'envato-',
+		'fbook',
+		'facebook',
+		'fb-',
+		'fb-messenger',
+		'fedex-',
+		'feedburner',
+		'firefox-',
+		'fontawesome-',
+		'font-awesome-',
+		'ganalytics-',
+		'gberg',
+		'github-',
+		'givewp-',
+		'google-',
+		'googlebot-',
+		'googles-',
+		'gravity-form-',
+		'gravity-forms-',
+		'gravityforms-',
+		'gtmetrix-',
+		'gutenberg',
+		'guten-',
+		'hubspot-',
+		'ig-',
+		'insta-',
+		'instagram',
+		'internet-explorer-',
+		'ios-',
+		'jetpack-',
+		'macintosh-',
+		'macos-',
+		'mailchimp-',
+		'microsoft-',
+		'ninja-forms-',
+		'oculus',
+		'onlyfans-',
+		'only-fans-',
+		'opera-',
+		'paddle-',
+		'paypal-',
+		'pinterest-',
+		'plugin',
+		'skype-',
+		'stripe-',
+		'tiktok-',
+		'tik-tok-',
+		'trustpilot',
+		'twitch-',
+		'twitter-',
+		'tweet',
+		'ups-',
+		'usps-',
+		'vvhatsapp',
+		'vvcommerce',
+		'vva-',
+		'vvoo',
+		'wa-',
+		'webpush-vn',
+		'wh4tsapps',
+		'whatsapp',
+		'whats-app',
+		'watson',
+		'windows-',
+		'wocommerce',
+		'woocom-',
+		'woocommerce',  // technically ending with '-for-woocommerce' is allowed.
+		'woocomerce',
+		'woo-commerce',
+		'woo-',
+		'wo-',
+		'wordpress',
+		'wordpess',
+		'wpress',
+		'wp-',
+		'wp-mail-smtp-',
+		'yandex-',
+		'yahoo-',
+		'yoast',
+		'youtube-',
+		'you-tube-',
+	);
+
+	// Domains from which exceptions would be accepted.
+	const TRADEMARK_EXCEPTIONS = array(
+		'adobe.com'             => array( 'adobe' ),
+		'automattic.com'        => array( 'akismet', 'akismet-', 'jetpack', 'jetpack-', 'wordpress', 'wp-', 'woo', 'woo-', 'woocommerce', 'woocommerce-' ),
+		'facebook.com'          => array( 'facebook', 'instagram', 'oculus', 'whatsapp' ),
+		'support.microsoft.com' => array( 'bing-', 'microsoft-' ),
+		'trustpilot.com'        => array( 'trustpilot' ),
+		'microsoft.com'         => array( 'bing-', 'microsoft-' ),
+		'yandex-team.ru'        => array( 'yandex' ),
+		'yoast.com'             => array( 'yoast' ),
+		'opera.com'             => array( 'opera-' ),
+		'adobe.com'             => array( 'adobe-' ),
+	);
+
+	// Trademarks that are allowed as 'for-whatever' ONLY.
+	const FOR_USE_EXCEPTIONS = array(
+		'woocommerce',
+	);
+
+	// Commonly used 'combo' names (to prevent things like 'woopress').
+	const PORTMANTEAUS = array(
+		'woo',
+	);
+
+	public function check_readme() {
+		$preamble = __( 'Error: The readme name includes a restricted term.', 'wporg-plugins' );
+
+		return $this->_trademark_check( $this->readme->name ?? false, $preamble );
+	}
+
+	public function check_plugin_name() {
+		$preamble = __( 'Error: The plugin name includes a restricted term.', 'wporg-plugins' );
+
+		return $this->_trademark_check( $this->headers['Name'] ?? false, $preamble );
+	}
+
+	public function check_plugin_slug() {
+		$preamble = __( 'Error: The plugin slug includes a restricted term.', 'wporg-plugins' );
+
+		return $this->_trademark_check( $this->slug ?? false, $preamble );
+	}
```

Comment:
This should not run for not-new plugins on WordPress.org, something like the below.

```suggestion
	public function check_plugin_slug() {
		if ( $this->post && ! in_array( $this->post->post_status, [ 'new', 'pending' ], true ) ) {
			// Do not check the slug for published/approved plugins. This check is only for new plugins and when being used in a local development environment.
			return;
		}

		$preamble = __( 'Error: The plugin slug includes a restricted term.', 'wporg-plugins' );

		return $this->_trademark_check( $this->slug ?? false, $preamble );
	}
```

Reply:
Resolved